### DISCUSSION TITLE
When is Microsoft.Win32.Registry 6.0.0 releasing

### DISCUSSION THREAD

**Discussion by ChronosSF** [dotnet/runtime#62055] [Q&A] [answered]
Hello,

We have a library with a dependency on Microsoft.Win32.Registry that we want to release as part of our product targeting .net 6, however, it seems this package has been left in a prerelease state for .net 6 for quite some months now.

Do you have a release date for this package 6.0.0 version?

**Comment by Joe4evr** [comment, marked as answer]
You shouldn't need an extra package. If you set `<TargetFramework>net6.0-windows</TargetFramework>` in your .csproj, you should get access to those types.

**Reply by huoyaoyuan** [reply]
See https://github.com/dotnet/runtime/issues/62009

**Reply by ChronosSF** [reply]
Thanks!

**Reply by ndepend** [reply]
@Joe4evr What about if the project is netstandard2.0 ? It used to reference Microsoft.Win32.Registry 5.0.0 seamlessly.

![image](https://user-images.githubusercontent.com/511445/188673594-5480ceb7-3bf9-477c-9b0b-ac66e395ba4c.png)


```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <!--<TargetFramework>net6.0-windows</TargetFramework>-->
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Win32.Registry" Version="6.0.0" />
  </ItemGroup>
</Project>
```

**Reply by huoyaoyuan** [reply]
Just continue to use the 5.0.0 version forever. There won't be a new package version for older frameworks.
When the netstandard library is used in projects targeting newer framework, the newer inbox version would be picked.

**Reply by ndepend** [reply]
@huoyaoyuan indeed it works, this is weird but it works. Thanks so much 
![image](https://user-images.githubusercontent.com/511445/188798286-9aaa3a2a-ffae-48f8-8449-c73cac965e96.png)

**Comment by danmoseley** [comment]
@ViktorHofer  in case he has more to add

**Reply by ViktorHofer** [reply]
Thanks for the ping. Others already provided most of the information. To summarize, we don't produce a package for Microsoft.Win32.Registy any longer as the library became part of the `Microsoft.NETCore.App` shared framework. If you target .NETStandard and want to use Registry APIs, feel free to use the existing stable 5.0.0 version of the package.

This change was made as part of .NET 6 and you should be able to use the Registry API without referencing a package when targeting net6.0 or higher.

**Reply by ndepend** [reply]
@ViktorHofer  thanks for the feedback, our of curiosity, how does the `net60 / net70` runtime accepts a reference from `netstandard2.0` to `Microsoft.Win32.Registy 5.0.0` (which is compiled against `.NET Framework v4.61 `AFAIK)  and translates it to calls to `net60 / net70` equivalent registry types?

**Reply by ViktorHofer** [reply]
The assembly version of the netstandard2.0/net461 asset in the Microsoft.Win32.Registry 5.0.0 package is 5.0.0.x. The assembly version of the Microsoft.Win32.Registry.dll that is part of the shared framework is 6.0.0.0 for the .NET 6 and 7.0.0.0 for the .NET 7 shared framework. The "conflict resolution" task is msbuild makes sure to pick the right assembly based on the assembly and file version.

In the example of .NET 6, the 6.0.0.0 Microsoft.Win32.Registry.dll assembly from the shared framework wins over the 5.0.0.0 netstandard2.0/Microsoft.Win32.Registry.dll from the nuget package.

**Comment by major-mayer** [comment]
FYI: In .NET 10 NuGet removes transitive dependencies to Microsoft.Win32.Registry package automatically from the dependency graph, because package pruning is enabled by default: https://devblogs.microsoft.com/dotnet/nuget-package-pruning-in-dotnet-10/
